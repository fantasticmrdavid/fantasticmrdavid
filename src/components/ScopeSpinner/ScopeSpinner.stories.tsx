import React from "react";
import { ScopeSpinner } from "./ScopeSpinner";
import { ComponentStory, Meta } from "@storybook/react";

export default {
  title: "ScopeSpinner",
  component: ScopeSpinner
} as Meta

const Template: ComponentStory<typeof ScopeSpinner> = () => <div style={{ margin: "10em" }}><ScopeSpinner /></div>;

export const Default = Template.bind({});
