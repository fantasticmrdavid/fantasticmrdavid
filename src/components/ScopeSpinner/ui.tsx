import React, { memo } from 'react';
import Ring from './Ring';
import * as styles from './styles';

const {
  Container,
} = styles;

const RingList = [
  {
    size: 210,
    rotateDirection: 'cc',
    strokeWidth: 12,
    minSlice: 65,
    maxSlice: 90,
    speed: 2.5,
  },
  {
    size: 200,
    strokeWidth: 4,
    minSlice: 100,
    maxSlice: 100,
    speed: 2.5,
  },
  {
    size: 150,
    strokeWidth: 10,
    rotateDirection: 'cc',
    minSlice: 55,
    maxSlice: 90,
    speed: 1,
  },
  {
    size: 150,
    strokeWidth: 2,
    minSlice: 100,
    maxSlice: 100,
    speed: 1,
  },
  {
    size: 120,
    rotateDirection: 'cc',
    strokeWidth: 20,
    minSlice: 45,
    maxSlice: 50,
    speed: 1,
  },
  {
    size: 100,
    strokeWidth: 3,
    rotateDirection: 'cc',
    minSlice: 70,
    maxSlice: 90,
    speed: 0.5,
  },
  {
    size: 100,
    strokeWidth: 8,
    rotateDirection: 'cc',
    minSlice: 30,
    maxSlice: 30,
    speed: 0.5,
  },
];

export default memo(
  () => (
    <Container>
      { RingList.map((r: any, i: number) => <Ring {...r} key={`ring_${i.toString()}`} />) }
    </Container>
  ),
);
