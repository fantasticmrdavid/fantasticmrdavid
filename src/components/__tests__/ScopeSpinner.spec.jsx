import React from 'react';
import renderer from 'react-test-renderer';
import ScopeSpinner from 'components/ScopeSpinner';

describe('Component -> ScopeSpinner', () => {
  it('should render the ScopeSpinner without throwing an error', () => {
    expect(renderer.create(<ScopeSpinner />)).toMatchSnapshot();
  });
});
