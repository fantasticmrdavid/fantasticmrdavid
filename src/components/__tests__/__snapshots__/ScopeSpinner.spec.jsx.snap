// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> ScopeSpinner should render the ScopeSpinner without throwing an error 1`] = `
<div
  className="styles__Container-sc-5fp6d1-0 gaARso"
>
  <div
    className="styles__Container-sc-1pex0aj-0 iPixZz"
    size={210}
    speed={2.5}
  >
    <div
      className="styles__Mask-sc-1pex0aj-2 YWAdm"
      speed={2.5}
    >
      <div
        className="styles__Ring-sc-1pex0aj-1 hZttlr"
        size={210}
        strokeWidth={12}
      />
    </div>
  </div>
  <div
    className="styles__Container-sc-1pex0aj-0 bBZVEC"
    size={200}
    speed={2.5}
  >
    <div
      className="styles__Mask-sc-1pex0aj-2 bFOSlo"
      speed={2.5}
    >
      <div
        className="styles__Ring-sc-1pex0aj-1 iiWvvC"
        size={200}
        strokeWidth={4}
      />
    </div>
  </div>
  <div
    className="styles__Container-sc-1pex0aj-0 iXnjYc"
    size={150}
    speed={1}
  >
    <div
      className="styles__Mask-sc-1pex0aj-2 ldFsQu"
      speed={1}
    >
      <div
        className="styles__Ring-sc-1pex0aj-1 cjwfle"
        size={150}
        strokeWidth={10}
      />
    </div>
  </div>
  <div
    className="styles__Container-sc-1pex0aj-0 iUEZIG"
    size={150}
    speed={1}
  >
    <div
      className="styles__Mask-sc-1pex0aj-2 iptFWs"
      speed={1}
    >
      <div
        className="styles__Ring-sc-1pex0aj-1 jPFpay"
        size={150}
        strokeWidth={2}
      />
    </div>
  </div>
  <div
    className="styles__Container-sc-1pex0aj-0 cwGLCp"
    size={120}
    speed={1}
  >
    <div
      className="styles__Mask-sc-1pex0aj-2 fvXegV"
      speed={1}
    >
      <div
        className="styles__Ring-sc-1pex0aj-1 XldDq"
        size={120}
        strokeWidth={20}
      />
    </div>
  </div>
  <div
    className="styles__Container-sc-1pex0aj-0 fPKfuu"
    size={100}
    speed={0.5}
  >
    <div
      className="styles__Mask-sc-1pex0aj-2 cVhCxa"
      speed={0.5}
    >
      <div
        className="styles__Ring-sc-1pex0aj-1 blYFTr"
        size={100}
        strokeWidth={3}
      />
    </div>
  </div>
  <div
    className="styles__Container-sc-1pex0aj-0 fPKfuu"
    size={100}
    speed={0.5}
  >
    <div
      className="styles__Mask-sc-1pex0aj-2 duOTjR"
      speed={0.5}
    >
      <div
        className="styles__Ring-sc-1pex0aj-1 gyDhAj"
        size={100}
        strokeWidth={8}
      />
    </div>
  </div>
</div>
`;
